src/Microsoft.Extensions.Caching.Memory/MemoryCacheServiceCollectionExtensions.cs

```diff
@@ -80,7 +80,44 @@ public static IServiceCollection AddDistributedMemoryCache(this IServiceCollecti
                 throw new ArgumentNullException(nameof(services));
             }
 
-            services.TryAddSingleton<IDistributedCache>(new MemoryDistributedCache(new MemoryCache(new MemoryCacheOptions())));
+            services.AddOptions();
+            services.TryAdd(ServiceDescriptor.Singleton<IDistributedCache, MemoryDistributedCache>());
+
+            return services;
+        }
+
+        /// <summary>
+        /// Adds a default implementation of <see cref="IDistributedCache"/> that stores items in memory
+        /// to the <see cref="IServiceCollection" />. Frameworks that require a distributed cache to work
+        /// can safely add this dependency as part of their dependency list to ensure that there is at least
+        /// one implementation available.
+        /// </summary>
+        /// <remarks>
+        /// <see cref="AddDistributedMemoryCache(IServiceCollection)"/> should only be used in single
+        /// server scenarios as this cache stores items in memory and doesn't expand across multiple machines.
+        /// For those scenarios it is recommended to use a proper distributed cache that can expand across
+        /// multiple machines.
+        /// </remarks>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <param name="setupAction">
+        /// The <see cref="Action{MemoryDistributedCacheOptions}"/> to configure the provided <see cref="MemoryDistributedCacheOptions"/>.
+        /// </param>
+        /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+        public static IServiceCollection AddDistributedMemoryCache(this IServiceCollection services, Action<MemoryDistributedCacheOptions> setupAction)
```

Comment:
This is ok. Both names are already used publicly, you can't change it without breaking APIs.